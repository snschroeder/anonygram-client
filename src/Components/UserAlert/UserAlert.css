.notifications-container {
  background-color: lightcoral;
  width: 80%;
  margin: 0 auto;
  position: sticky;
  position: -webkit-sticky;
  text-align: center;
  margin-top: 5px;
  margin-bottom: 5px;
  padding-top: 2px;
  padding-bottom: 2px;
}