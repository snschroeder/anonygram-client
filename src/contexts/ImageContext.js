import React, { useContext } from 'react';
export const useImageContext = () => useContext(ImageContext);
const defaultValues = {
  //vars in ABC order
  alert: '',
  debounce: null,
  error: null,
  images: [],
  morePagesAvail: null,
  newContentLoaded: null, 
  page: null,
  sort: ['new', 'top'],
  user: null,
  userLocation: {},
  //funcs in ABC order
  clearAlert: () => {},
  clearError: () => {},
  handleDelete: () => {},
  incrementUpvotes: () => {},
  resetState: () => {},
  setAlert: () => {},
  setDebounce: () => {},
  setError: () => {},
  setMorePagesAvail: () => {},
  setNewContentLoaded: () => {},
  setPage: () => {},
  setView: () => {},
}
const ImageContext = React.createContext(defaultValues)

export default ImageContext;
