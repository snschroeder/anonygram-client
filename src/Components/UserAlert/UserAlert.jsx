import React from 'react';
import { useImageContext } from '../../contexts/ImageContext';
import './UserAlert.css';

export default function UserAlert() {

  const context = useImageContext();

  const { alert } = context;

  return (
    <>
    {alert && 
    <div className="notifications-container">
      <p className="message">{alert}</p>
      </div>}
    </>
  )
}
