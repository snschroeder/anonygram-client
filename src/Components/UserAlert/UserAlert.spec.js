import React from 'react';
import ReactDOM from 'react-dom';
import UserAlert from './UserAlert';
import * as ImageContext from '../../contexts/ImageContext'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('UserAlert component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserAlert />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders alert when set in context', () => {
    const contextValues = {
      alert: 'Alert!'
    }
    jest
      .spyOn(ImageContext, 'useImageContext')
      .mockImplementation(() => contextValues)
    const wrapper = shallow(<UserAlert/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
