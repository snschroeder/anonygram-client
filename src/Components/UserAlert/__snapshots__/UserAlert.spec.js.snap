// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAlert component renders alert when set in context 1`] = `
<Fragment>
  <div
    className="notifications-container"
  >
    <p
      className="message"
    >
      Alert!
    </p>
  </div>
</Fragment>
`;
